mcp/client.go

```diff
@@ -174,6 +174,59 @@ func (cs *ClientSession) ID() string {
 	return cs.mcpConn.SessionID()
 }
 
+// ServerCapabilities returns a copy of the server capabilities obtained during initialization.
+// If the session has not been initialized or capabilities are not available, it returns nil.
+func (cs *ClientSession) ServerCapabilities() *ServerCapabilities {
```

Comment:
I wonder if we can just use an exported field. The doc will say not to modify it, but even if someone does, they're only hurting themselves, as far as I can tell. 

Or if we're not sure, let's use the function, but not bother copying until we discover a case where it could matter. But really I think the field suffices.

/cc @samthanawalla @findleyr

Reply:
I agree with not copying right now.

I think we should go with the cs.ServerCapabilities() function because cs.InitializeResult.ServerCapabilities is a bit cumbersome/less obvious to use. Plus, like you said, we can add copying to ServerCapabilities later.